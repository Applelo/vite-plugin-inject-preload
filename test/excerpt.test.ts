import { resolve } from 'path'
import { build } from 'vite'
import { describe, expect, it } from 'vitest'
import type { OutputAsset, RollupOutput } from 'rollup'
import VitePluginInjectPreload from './../src/index'
import type { Options } from './../src/index'

const configs: Record<string, Options> = {
  injectBottom: {
    files: [
      {
        match: /Roboto-[a-zA-Z]*\.[a-z-0-9]*.woff2$/
      },
      {
        match: /lazy.[a-z-0-9]*.(css|js)$/
      }
    ],
    injectTo: 'head'
  },
  customAttributes: {
    files: [
      {
        match: /Roboto-[a-zA-Z]*\.[a-z-0-9]*.woff2$/,
        attributes: {
          as: 'font',
          crossorigin: 'anonymous',
          'data-font': 'Roboto'
        }
      }
    ]
  },
  auto: {
    files: [
      {
        match: /Roboto-[a-zA-Z]*\.[a-z-0-9]*.woff2$/
      },
      {
        match: /lazy.[a-z-0-9]*.(css|js)$/
      }
    ]
  }
}

const buildVite = async (config: Options) => {
  const { output } = (await build({
    root: resolve(__dirname, './../demo'),
    plugins: [VitePluginInjectPreload(config)]
  })) as RollupOutput

  const { source: indexSource } = output.find(
    item => item.fileName === 'index.html'
  ) as OutputAsset

  return indexSource.toString()
}

describe('excerpt', () => {
  for (const key in configs) {
    if (Object.prototype.hasOwnProperty.call(configs, key)) {
      const config = configs[key] as Options
      it(`test ${key}`, async () => {
        const output = await buildVite(config)
        expect(output).toMatchSnapshot()
      })
    }
  }
})
