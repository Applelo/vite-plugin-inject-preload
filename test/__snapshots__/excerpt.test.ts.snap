// Vitest Snapshot v1

exports[`excerpt > test auto 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <link rel=\\"preload\\" href=\\"/assets/Roboto-Italic.7d6fc55f.woff2\\" type=\\"font/woff2\\" as=\\"font\\">
    <link rel=\\"preload\\" href=\\"/assets/Roboto-Regular.a9876de8.woff2\\" type=\\"font/woff2\\" as=\\"font\\">
    <link rel=\\"preload\\" href=\\"/assets/lazy.53d49301.js\\" type=\\"application/javascript\\" as=\\"script\\">
    <link rel=\\"preload\\" href=\\"/assets/lazy.b3f56c2d.css\\" type=\\"text/css\\" as=\\"style\\">

    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>Vite Plugin Inject Preload</title>
    <!--__vite-plugin-inject-preload__-->
    <script type=\\"module\\" crossorigin src=\\"/assets/index.496d25c6.js\\"></script>
    <link rel=\\"stylesheet\\" href=\\"/assets/index.a44c53f4.css\\">
  </head>
  <body>
    <h1>Vite Plugin Inject Preload</h1>
    <a href=\\"page.html\\">Go to the page (test multi page generation)</a>
    
  </body>
</html>
"
`;

exports[`excerpt > test customAttributes 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <link rel=\\"preload\\" href=\\"/assets/Roboto-Italic.7d6fc55f.woff2\\" type=\\"font/woff2\\" as=\\"font\\" crossorigin=\\"anonymous\\" data-font=\\"Roboto\\">
    <link rel=\\"preload\\" href=\\"/assets/Roboto-Regular.a9876de8.woff2\\" type=\\"font/woff2\\" as=\\"font\\" crossorigin=\\"anonymous\\" data-font=\\"Roboto\\">

    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>Vite Plugin Inject Preload</title>
    <!--__vite-plugin-inject-preload__-->
    <script type=\\"module\\" crossorigin src=\\"/assets/index.496d25c6.js\\"></script>
    <link rel=\\"stylesheet\\" href=\\"/assets/index.a44c53f4.css\\">
  </head>
  <body>
    <h1>Vite Plugin Inject Preload</h1>
    <a href=\\"page.html\\">Go to the page (test multi page generation)</a>
    
  </body>
</html>
"
`;

exports[`excerpt > test customPosition 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>Vite Plugin Inject Preload</title>
    <link rel=\\"preload\\" href=\\"/assets/Roboto-Italic.7d6fc55f.woff2\\" type=\\"font/woff2\\" as=\\"font\\" data-vite-plugin-inject-preload>
    <link rel=\\"preload\\" href=\\"/assets/Roboto-Regular.a9876de8.woff2\\" type=\\"font/woff2\\" as=\\"font\\" data-vite-plugin-inject-preload>
    <link rel=\\"preload\\" href=\\"/assets/lazy.53d49301.js\\" type=\\"application/javascript\\" as=\\"script\\">
    <link rel=\\"preload\\" href=\\"/assets/lazy.b3f56c2d.css\\" type=\\"text/css\\" as=\\"style\\">

    <script type=\\"module\\" crossorigin src=\\"/assets/index.496d25c6.js\\"></script>
    <link rel=\\"stylesheet\\" href=\\"/assets/index.a44c53f4.css\\">
  </head>
  <body>
    <h1>Vite Plugin Inject Preload</h1>
    <a href=\\"page.html\\">Go to the page (test multi page generation)</a>
    
  </body>
</html>
"
`;

exports[`excerpt > test injectBottom 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>Vite Plugin Inject Preload</title>
    <!--__vite-plugin-inject-preload__-->
    <script type=\\"module\\" crossorigin src=\\"/assets/index.496d25c6.js\\"></script>
    <link rel=\\"stylesheet\\" href=\\"/assets/index.a44c53f4.css\\">
    <link rel=\\"preload\\" href=\\"/assets/Roboto-Italic.7d6fc55f.woff2\\" type=\\"font/woff2\\" as=\\"font\\">
    <link rel=\\"preload\\" href=\\"/assets/Roboto-Regular.a9876de8.woff2\\" type=\\"font/woff2\\" as=\\"font\\">
    <link rel=\\"preload\\" href=\\"/assets/lazy.53d49301.js\\" type=\\"application/javascript\\" as=\\"script\\">
    <link rel=\\"preload\\" href=\\"/assets/lazy.b3f56c2d.css\\" type=\\"text/css\\" as=\\"style\\">
  </head>
  <body>
    <h1>Vite Plugin Inject Preload</h1>
    <a href=\\"page.html\\">Go to the page (test multi page generation)</a>
    
  </body>
</html>
"
`;

exports[`excerpt > test wrongAttributes 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <link rel=\\"preload\\" href=\\"./yolo.woff2\\" type=\\"font/woff2\\" as=\\"font\\">
    <link rel=\\"preload\\" href=\\"./yolo.woff2\\" type=\\"font/woff2\\" as=\\"font\\">

    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>Vite Plugin Inject Preload</title>
    <!--__vite-plugin-inject-preload__-->
    <script type=\\"module\\" crossorigin src=\\"/assets/index.496d25c6.js\\"></script>
    <link rel=\\"stylesheet\\" href=\\"/assets/index.a44c53f4.css\\">
  </head>
  <body>
    <h1>Vite Plugin Inject Preload</h1>
    <a href=\\"page.html\\">Go to the page (test multi page generation)</a>
    
  </body>
</html>
"
`;
